File: BBKBootcampSocial.Core/DTOs/Comment/CommentDTO.cs
﻿using System.Collections.Generic;

namespace BBKBootcampSocial.Core.DTOs.Comment
{
    public class CommentDTO
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string ProfilePic { get; set; }
        public long UserId { get; set; }
        public string Text { get; set; }
        public int LikeCount { get; set; }
        public long Id { get; set; }
        public long PostId { get; set; }
        public long? ParentId { get; set; }
        public IEnumerable<CommentDTO> Replies { get; set; }
    }
}

File: ClientSide/src/app/DTOs/CommentDTOs/CommentDTO.ts
export class CommentDTO {
    constructor(
        public id : number,
        public firstName: string,
        public lastName: string,
        public profilePic: string,
        public userId: number,
        public text: string,
        public likeCount: number,
        public postId: number,
        public parentId: number,
        public replies: CommentDTO[]
    ) { }
}